src/preact/base-element.js

```diff
@@ -369,22 +369,6 @@ export class PreactBaseElement extends BaseElement {
     }
   }
 
-  /** @override */
-  attemptChangeHeight(newHeight) {
-    return super.attemptChangeHeight(newHeight).catch((e) => {
-      // It's okay to disable this lint rule since we check that the restricted
-      // method exists.
-      // eslint-disable-next-line local/restrict-this-access
-      if (this.getOverflowElement && !this.getOverflowElement()) {
```

Comment:
It checks if `this.getOverFlowElement` exists before calling precisely to guard against non-AMP cases: https://github.com/ampproject/amphtml/pull/35027#discussion_r665751933

If removing is preferred we should re-add into the AMP layers of affected components (`amp-twitter`, `amp-facebook`, `amp-instagram`) and [update the documentation](https://github.com/ampproject/amphtml/blob/main/docs/building-a-bento-iframe-component.md) for future components.

Reply:
@caroqliu thanks for the context on why this was added. Based on [this comment](https://github.com/ampproject/amphtml/pull/35566#discussion_r692501850) it seems like we should not include any AMP specific code in preact base element. I'm ok either way and leave it up to you and @jridgewell to decide 🙂